src/hooks/useCopyToClipboard.ts

```diff
@@ -1,29 +1,37 @@
 import {useEffect, useState} from 'react';
 
+import useSecureContext from '@hooks/useSecureContext';
+
 type CopyToClipboardOptions = {
+  filename?: string;
   timeoutInMs?: number;
   notificationDuration?: number;
 };
 
 export const useCopyToClipboard = (textToCopy: string, options: CopyToClipboardOptions = {timeoutInMs: 2000}) => {
   const {timeoutInMs} = options;
 
-  const [isCopied, setCopyToClipboardState] = useState(false);
+  const [isProcessed, setProcessed] = useState(false);
+  const isSecureContext = useSecureContext();
+
+  const copyToClipBoard = () => {
+    navigator.clipboard.writeText(textToCopy);
+    return setTimeout(() => {
+      setProcessed(false);
+    }, timeoutInMs);
+  };
 
   useEffect(() => {
     let timeout: NodeJS.Timeout;
 
-    if (isCopied) {
-      navigator.clipboard.writeText(textToCopy);
-      timeout = setTimeout(() => {
-        setCopyToClipboardState(false);
-      }, timeoutInMs);
+    if (isProcessed && isSecureContext) {
+      timeout = copyToClipBoard();
     }
 
     return () => {
       clearTimeout(timeout);
     };
-  }, [isCopied]);
+  }, [isProcessed, window]);
```

Comment:
Just wondering why we need `window` in the dependency array

Reply:
To be honest I'm not entirely sure. I think it might be redundant.

Reply:
Removed